akka-actor/src/main/scala/akka/util/MessageBuffer.scala

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.util
+
+import akka.actor.ActorRef
+
+/**
+ * A single producer single consumer mutable message buffer
+ */
+final class MessageBuffer private (
```

Comment:
I mean, should this class be public API at all or should we keep it internal?
If public it must have proper Java API for everything.
I vote for making it internal.